sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/FullFidelityChangeFeedProcessorBuilder.java

```diff
@@ -0,0 +1,179 @@
+    }
+
+    /**
+     * Sets an existing {@link CosmosAsyncContainer} to be used to read from the leases container.
+     *
+     * @param leaseContainer the instance of {@link CosmosAsyncContainer} to use.
+     * @return current Builder.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public FullFidelityChangeFeedProcessorBuilder leaseContainer(CosmosAsyncContainer leaseContainer) {
+        this.leaseContainer = leaseContainer;
+
+        return this;
+    }
+
+    /**
+     * Sets a consumer function which will be called to process changes.
+     *
+     * <!-- src_embed com.azure.cosmos.fullFidelityChangeFeedProcessor.handleChanges -->
+     * <pre>
+     * .handleChanges&#40;docs -&gt; &#123;
+     *     for &#40;JsonNode item : docs&#41; &#123;
+     *         &#47;&#47; Implementation for handling and processing of each JsonNode item goes here
+     *     &#125;
+     * &#125;&#41;
+     * </pre>
+     * <!-- end com.azure.cosmos.fullFidelityChangeFeedProcessor.handleChanges -->
+     *
+     * @param consumer the {@link Consumer} to call for handling the feeds.
+     * @return current Builder.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public FullFidelityChangeFeedProcessorBuilder handleChanges(Consumer<List<ChangeFeedProcessorItem>> consumer) {
+        this.consumer = consumer;
+
+        return this;
+    }
+
+    /**
+     * Sets the {@link ChangeFeedProcessorOptions} to be used. Unless specifically set the default values that will be
+     * used are:
+     * <ul>
+     * <li>maximum items per page or FeedResponse: 100</li>
+     * <li>lease renew interval: 17 seconds</li>
+     * <li>lease acquire interval: 13 seconds</li>
+     * <li>lease expiration interval: 60 seconds</li>
+     * <li>feed poll delay: 5 seconds</li>
+     * <li>maximum scale count: unlimited</li>
+     * </ul>
+     *
+     * @param changeFeedProcessorOptions the change feed processor options to use.
+     * @return current Builder.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public FullFidelityChangeFeedProcessorBuilder options(ChangeFeedProcessorOptions changeFeedProcessorOptions) {
+        this.changeFeedProcessorOptions = changeFeedProcessorOptions;
+
+        return this;
+    }
+
+    /**
+     * Builds a new instance of the {@link ChangeFeedProcessor} with the specified configuration.
+     *
+     * @return an instance of {@link ChangeFeedProcessor}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_34_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public ChangeFeedProcessor buildChangeFeedProcessor() {
+        if (hostName == null || hostName.isEmpty()) {
+            throw new IllegalArgumentException("hostName");
+        }
+        if (feedContainer == null) {
+            throw new IllegalArgumentException("feedContainer");
```

Comment:
Given that Java doesn't have proper pendent to ArgumentNullException I guess you'll need to provide some more ocntext in the exception message. It isn't obvious that this exception is triggered by not calling some builder api